### ISSUE TITLE
planemo shed_test fails

### ISSUE THREAD

**Issue by lecorguille** [galaxyproject/planemo#424]
Hi all,

All along the year, I promote functional tests and planemo through trainings.
But, but, unfortunatly, my own tools have difficulties to pass using planemo shed_test.

I have several tools in queue waiting to pass. And this one is the simplest:
TS: https://toolshed.g2.bx.psu.edu/view/lecorguille/anova/2433c6fa4de2
GitHub: https://github.com/workflow4metabolomics/anova/tree/master/galaxy/stat_anova

I hope that for you it's obvious that it's a problem on my setup (OS, Planemo version), or on my usage (option) or in my wrapper syntax or about me.

There are not so bad since there pass the **planemo test** 

```
source ~/galaxy/deps/R/3.1.2/iuc/package_r_3_1_2/1ca39eb16186/env.sh
source ~/galaxy/deps/batch/1.1-4/lecorguille/package_r_batch_1_1_4/f7e3fa941a02/env.sh 
planemo test --install_galaxy

#All 1 test(s) executed passed.
#abims_anova[0]: passed
```

But **planemo shed_test** aren't so cooperative. 
I launch them on 2 systems and get 2 different results: some  nodes (qlogin) of our HPC under CentOS6 and a new VM under Unbuntu.

- CentOS6

```
planemo shed_test --install_galaxy
[...]
Installing repositories - this may take some time...
Traceback (most recent call last):
  File "/w/galaxy/galaxy4gwen/planemoenv/bin/planemo", line 11, in <module>
    sys.exit(planemo())
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/planemo/commands/cmd_shed_test.py", line 39, in cli
    with galaxy_serve.shed_serve(ctx, install_args_list, **kwds) as config:
  File "/opt/6.x/python-2.7.2/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/planemo/galaxy_serve.py", line 58, in shed_serve
    config.wait_for_all_installed()
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/planemo/galaxy_config.py", line 361, in wait_for_all_installed
    wait_on(ready, "galaxy tool installation", timeout=60 * 60 * 1)
  File "/w/galaxy/galaxy4gwen/planemoenv/lib/python2.7/site-packages/planemo/io.py", line 225, in wait_on
    raise Exception(message)
Exception: Timed out waiting on galaxy tool installation.
```

Although, the dependenties seem correctly installed in deps and in shed_tools
```
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 21:58 libxml2
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:00 libpng
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:01 ncurses
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:03 zlib
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:03 freetype
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:04 fontconfig
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:05 readline
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:06 pixman
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:08 cairo
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:17 R
drwxr-xr-x  3 lecorguille sib 4096 Mar 28 22:22 batch
```

- Ubuntu in a VM

```
planemo shed_test --install_galaxy

galaxy.jobs.command_factory INFO 2016-03-28 23:42:26,371 Built script [/tmp/tmpT82UaU/job_working_directory/000/1/tool_script.sh] for tool command[python /tmp/tmpT82UaU/galaxy-dev/tools/data_source/upload.py /tmp/tmpT82UaU/galaxy-dev /tmp/tmpT82UaU/tmp/tmp2NJiqC /tmp/tmpT82UaU/tmp/tmpmmdTrh 1:/tmp/tmpT82UaU/job_working_directory/000/1/dataset_1_files:/tmp/tmpT82UaU/files/000/dataset_1.dat]
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:26,471 "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-28 23:42:26,475 Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:26,599 "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-28 23:42:26,606 Starting new HTTP connection (1): localhost
galaxy.jobs.runners ERROR 2016-03-28 23:42:26,630 (1) Failure preparing job
Traceback (most recent call last):
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 170, in prepare_job
    include_work_dir_outputs=include_work_dir_outputs,
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 200, in build_command_line
    container=container
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/command_factory.py", line 78, in build_command
    __handle_metadata(commands_builder, job_wrapper, runner, remote_command_params)
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/command_factory.py", line 156, in __handle_metadata
    kwds={ 'overwrite': False }
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1617, in setup_external_metadata
    dependency_shell_commands = metadata_tool.build_dependency_shell_commands(job_directory=self.working_directory)
AttributeError: 'NoneType' object has no attribute 'build_dependency_shell_commands'
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:26,719 "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-28 23:42:26,723 Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:26,858 "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-28 23:42:26,860 Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:26,949 "GET /api/histories/2891970512fa2d5a/contents?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-28 23:42:26,951 Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:27,102 "GET /api/histories/2891970512fa2d5a/contents/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool INFO 2016-03-28 23:42:27,104 Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool DEBUG 2016-03-28 23:42:27,314 "GET /api/histories/2891970512fa2d5a/contents/2891970512fa2d5a/provenance?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
ERROR

======================================================================
ERROR: Anova ( toolshed.g2.bx.psu.edu/repos/lecorguille/anova/abims_anova/1.1.3 ) > Test-1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/tmpT82UaU/galaxy-dev/test/functional/test_toolbox.py", line 289, in test_tool
    self.do_it( td )
  File "/tmp/tmpT82UaU/galaxy-dev/test/functional/test_toolbox.py", line 38, in do_it
    stage_data_in_history( galaxy_interactor, testdef.test_data(), test_history, shed_tool_id )
  File "/tmp/tmpT82UaU/galaxy-dev/test/base/interactor.py", line 41, in stage_data_in_history
    upload_wait()
  File "/tmp/tmpT82UaU/galaxy-dev/test/base/interactor.py", line 278, in wait
    while not self.__history_ready( history_id ):
  File "/tmp/tmpT82UaU/galaxy-dev/test/base/interactor.py", line 300, in __history_ready
    return self._state_ready( state, error_msg="History in error state." )
  File "/tmp/tmpT82UaU/galaxy-dev/test/base/interactor.py", line 365, in _state_ready
    raise Exception( error_msg )
Exception: History in error state.
-------------------- >> begin captured stdout << ---------------------
History with id 2891970512fa2d5a in error - summary of datasets in error below.
--------------------------------------
| 1 - dataMatrix.tsv (HID - NAME) 
| Dataset Blurb:
|  tool error
| Dataset Info:
|  failure preparing job
| Dataset Job Standard Output:
|  *Standard output was empty.*
| Dataset Job Standard Error:
|  *Standard error was empty.*
|
--------------------------------------

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/users?key=test_key HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/users HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/users/2891970512fa2d5a/api_key HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/histories HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.tools: DEBUG: Validated and populated state for tool request (10.333 ms)
galaxy.tools.actions.upload: DEBUG: Persisted uploads (15.713 ms)
galaxy.tools.actions.upload: DEBUG: Checked and cleaned uploads (141.882 ms)
galaxy.tools.actions.upload_common: INFO: tool upload1 created job id 1
galaxy.tools.actions.upload: DEBUG: Created upload job (94.523 ms)
galaxy.tools.execute: DEBUG: Tool [upload1] created job [1] (252.893 ms)
galaxy.tools.execute: DEBUG: Executed all jobs for tool request: (284.372 ms)
requests.packages.urllib3.connectionpool: DEBUG: "POST /api/tools HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.jobs: DEBUG: (1) Working directory for job is: /tmp/tmpT82UaU/job_working_directory/000/1
galaxy.jobs.handler: DEBUG: (1) Dispatching to local runner
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.jobs: DEBUG: (1) Persisting job destination (destination id: local:///)
galaxy.jobs.runners: DEBUG: Job [1] queued (87.048 ms)
galaxy.jobs.handler: INFO: (1) Job dispatched
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.tools.deps: DEBUG: Building dependency shell command for dependency 'samtools'
galaxy.tools.deps: DEBUG: Find dependency samtools version None
galaxy.tools.deps: DEBUG: Resolver tool_shed_packages returned <galaxy.tools.deps.resolvers.NullDependency object at 0x7fa8047bec90> (isnull? True)
galaxy.tools.deps: DEBUG: Resolver galaxy_packages returned <galaxy.tools.deps.resolvers.NullDependency object at 0x7fa8047bec90> (isnull? True)
galaxy.tools.deps: DEBUG: Resolver galaxy_packages returned <galaxy.tools.deps.resolvers.NullDependency object at 0x7fa8047bec90> (isnull? True)
galaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x7fa8047bec90> (isnull? True)
galaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x7fa8047bec90> (isnull? True)
galaxy.tools.deps: WARNING: Failed to resolve dependency on 'samtools', ignoring
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=38e25ec6039b6bbcf44d62cdcace27fb HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.jobs.command_factory: INFO: Built script [/tmp/tmpmuJQJs/job_working_directory/000/1/tool_script.sh] for tool command[python /tmp/tmpmuJQJs/galaxy-dev/tools/data_source/upload.py /tmp/tmpmuJQJs/galaxy-dev /tmp/tmpmuJQJs/tmp/tmpR86p38 /tmp/tmpmuJQJs/tmp/tmpDAqel8 1:/tmp/tmpmuJQJs/job_working_directory/000/1/dataset_1_files:/tmp/tmpmuJQJs/files/000/dataset_1.dat]
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
galaxy.jobs.runners: ERROR: (1) Failure preparing job
Traceback (most recent call last):
  File "/tmp/tmpmuJQJs/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 170, in prepare_job
    include_work_dir_outputs=include_work_dir_outputs,
  File "/tmp/tmpmuJQJs/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 200, in build_command_line
    container=container
  File "/tmp/tmpmuJQJs/galaxy-dev/lib/galaxy/jobs/command_factory.py", line 78, in build_command
    __handle_metadata(commands_builder, job_wrapper, runner, remote_command_params)
  File "/tmp/tmpmuJQJs/galaxy-dev/lib/galaxy/jobs/command_factory.py", line 156, in __handle_metadata
    kwds={ 'overwrite': False }
  File "/tmp/tmpmuJQJs/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1617, in setup_external_metadata
    dependency_shell_commands = metadata_tool.build_dependency_shell_commands(job_directory=self.working_directory)
AttributeError: 'NoneType' object has no attribute 'build_dependency_shell_commands'
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a/contents?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a/contents/2891970512fa2d5a?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
requests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost
requests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/2891970512fa2d5a/contents/2891970512fa2d5a/provenance?key=2a972784aab52c1aee132f0b28dbe2e0 HTTP/1.1" 200 None
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
XML: /tmp/tmpmuJQJs/xunit.xml
----------------------------------------------------------------------
Ran 1 test in 2.820s

FAILED (errors=1)
functional_tests.py INFO 2016-03-24 23:14:41,951 Shutting down
functional_tests.py INFO 2016-03-24 23:14:41,951 Shutting down embedded web server
functional_tests.py INFO 2016-03-24 23:14:41,953 Embedded web server stopped
functional_tests.py INFO 2016-03-24 23:14:41,954 Shutting down app
galaxy.jobs.handler INFO 2016-03-24 23:14:41,954 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-24 23:14:41,954 job handler queue stopped
galaxy.jobs.runners INFO 2016-03-24 23:14:41,954 TaskRunner: Sending stop signal to 2 worker threads
galaxy.jobs.runners INFO 2016-03-24 23:14:41,954 LocalRunner: Sending stop signal to 5 worker threads
galaxy.jobs.handler INFO 2016-03-24 23:14:41,957 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-24 23:14:41,957 job handler stop queue stopped
functional_tests.py INFO 2016-03-24 23:14:41,957 Embedded Universe application stopped
functional_tests.py INFO 2016-03-24 23:14:41,957 Cleaning up temporary files in /tmp/tmpB29yNz/tmp2kWemM
galaxy.jobs.handler INFO 2016-03-24 23:14:42,858 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-24 23:14:42,859 job handler queue stopped
galaxy.jobs.runners INFO 2016-03-24 23:14:42,859 TaskRunner: Sending stop signal to 2 worker threads
galaxy.jobs.runners INFO 2016-03-24 23:14:42,859 LocalRunner: Sending stop signal to 5 worker threads
galaxy.jobs.handler INFO 2016-03-24 23:14:42,860 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-24 23:14:42,860 job handler stop queue stopped
There were problems with 1 test(s) - out of 1 test(s) executed. See /home/lecorguille/script/git/anova/galaxy/stat_anova/tool_test_output.html for detailed breakdown.
```
Here, it's seem to come from the upload step.

**Note that some other team project pass the planemo shed_test under CentOS but fails with the same message on my Ubuntu setup.**

Many thanks by advance.

:beers: during GCC2016 for the guys who could help me to spot the source of my problem

**Comment by mvdbeek** [comment]
Hi @lecorguille ,

you're running into a timeout while installing the tool dependencies.
The timeout is set to 1h in https://github.com/galaxyproject/planemo/blob/master/planemo/galaxy_config.py#L361 

You are installing package_r_3_1_2 as a dependency, which has a huge dependency chain
(bigger than 3.2.1) where some packages are duplicated (devteam and iuc pixman, multiple versions of cairo, and so on).

There are a few things you can do: Increase the timeout in galaxy_config.py, to make sure you've actually got a chance to install the packages.
If your internet connection is as flaky as it is at the UPMC, you can try to use a cloud-instance.
Finally you could move to package_r_3_2_1, where some of the duplication is gone, but your best-bet is to try to get your packages into bioconda. That doesn't work for older galaxy installations though.

**Comment by bgruening** [comment]
Just as a side note we can add old dependencies into the IUC conda channel here: https://github.com/galaxyproject/conda-iuc

**Comment by jmchilton** [comment]
This is a Galaxy bug that should be fixed when https://github.com/galaxyproject/galaxy/pull/2024 reaches the master branch - you can target release_16.01 directly now and it should fix the problem.

**Comment by jmchilton** [comment]
Pushed out the fix to galaxyproject/galaxy master branch that was causing this traceback:

```
Traceback (most recent call last):
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 170, in prepare_job
    include_work_dir_outputs=include_work_dir_outputs,
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 200, in build_command_line
    container=container
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/command_factory.py", line 78, in build_command
    __handle_metadata(commands_builder, job_wrapper, runner, remote_command_params)
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/command_factory.py", line 156, in __handle_metadata
    kwds={ 'overwrite': False }
  File "/tmp/tmpT82UaU/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1617, in setup_external_metadata
    dependency_shell_commands = metadata_tool.build_dependency_shell_commands(job_directory=self.working_directory)
AttributeError: 'NoneType' object has no attribute 'build_dependency_shell_commands'
```

I guess this plus @mvdbeek's advice is enough to close this issue?

**Comment by lecorguille** [comment]
@mvdbeek About the timeout on CentOS, indeed, the deps installation take time but after 25min, all deps seem installed in /tmp/tmp*/deps. After that, I have to wait 35min (or more I didn't watch the clock) to get the timeout message. Thus, I don't know what it is waiting for. Mystery!

@mvdbeek About conda, sure I plan to look at that. But I want to pass those functional test before carry on some new features. I already get my accompt on bioconda :+1: 

@jmchilton Your fix come with a good timing for the Ubuntu case. Even if until now I didn't manage to use `planemo shed_test --install_galaxy --galaxy_branch "release_16.01"` (same error), I get better result using a local installation of Galaxy.

So now, my test seem to pass :clap: but unfortunately I didn't obtain the beautiful webpage with all in green. Instead, I get this minimal green box 

> ID: functional.test_toolbox.TestForTool_toolshed.g2.bx.psu.edu/repos/lecorguille/anova/abims_anova/1.1.3.test_tool_000000
> Description: Anova ( toolshed.g2.bx.psu.edu/repos/lecorguille/anova/abims_anova/1.1.3 ) > Test-1
> Status: success


And here is what I get on in logs.
```
[...]
ok

----------------------------------------------------------------------
Ran 1 test in 19.989s

OK
functional_tests.py INFO 2016-03-30 14:55:03,970 Shutting down
functional_tests.py INFO 2016-03-30 14:55:03,970 Shutting down embedded web server
functional_tests.py INFO 2016-03-30 14:55:03,974 Embedded web server stopped
functional_tests.py INFO 2016-03-30 14:55:03,975 Shutting down app
galaxy.jobs.handler INFO 2016-03-30 14:55:03,975 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-30 14:55:03,975 job handler queue stopped
galaxy.jobs.runners INFO 2016-03-30 14:55:03,975 TaskRunner: Sending stop signal to 2 worker threads
galaxy.jobs.runners INFO 2016-03-30 14:55:03,975 LocalRunner: Sending stop signal to 5 worker threads
galaxy.jobs.handler INFO 2016-03-30 14:55:03,979 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-30 14:55:03,979 job handler stop queue stopped
functional_tests.py INFO 2016-03-30 14:55:03,979 Embedded Universe application stopped
functional_tests.py INFO 2016-03-30 14:55:03,979 Cleaning up temporary files in /tmp/tmprNCezi/tmpuN3zn5
galaxy.jobs.handler INFO 2016-03-30 14:55:04,897 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-30 14:55:04,897 job handler queue stopped
galaxy.jobs.runners INFO 2016-03-30 14:55:04,897 TaskRunner: Sending stop signal to 2 worker threads
galaxy.jobs.runners INFO 2016-03-30 14:55:04,898 LocalRunner: Sending stop signal to 5 worker threads
galaxy.jobs.handler INFO 2016-03-30 14:55:04,898 sending stop signal to worker thread
galaxy.jobs.handler INFO 2016-03-30 14:55:04,898 job handler stop queue stopped
Traceback (most recent call last):
  File "/home/lecorguille/.venv_planemo/bin/planemo", line 11, in <module>
    sys.exit(planemo())
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/planemo/commands/cmd_shed_test.py", line 45, in cli
    **kwds
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/planemo/galaxy_test/actions.py", line 87, in run_in_config
    return_code,
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/planemo/galaxy_test/structures.py", line 166, in __init__
    sd = StructuredData(output_json_path)
  File "/home/lecorguille/.venv_planemo/local/lib/python2.7/site-packages/planemo/galaxy_test/structures.py", line 63, in __init__
    if not os.path.exists(json_path):
  File "/home/lecorguille/.venv_planemo/lib/python2.7/genericpath.py", line 18, in exists
    os.stat(path)
TypeError: coercing to Unicode: need string or buffer, NoneType found
```

But let me know if it's an another story that not concerning this Issue. Or if you are already aware of this problem somewhere.


Anyway, :beers: :beer: for you guys!

**Comment by jmchilton** [comment]
That is a baffling stack trace - it should never throw a stack trace there. Was there a warning anywhere in the planemo output containing a string like "Cannot locate json report option for tests - update"?

**Comment by lecorguille** [comment]
Hi,

I close this issue.
Between, update of planemo, improvement in my wrappers (thanks `planemo test` and conda), random cosmic events and because it's Friday, ... there isn't any issue anymore both on CentOS and Ubuntu.

Thanks. I will not forget your beers!
